teps/0129-multiple-tekton-instances-per-cluster.md

```diff
@@ -95,40 +95,181 @@ bundled in the Tekton webhook as well the remote resolution controller.
 
 ## Proposal
 
-<!--
-This is where we get down to the specifics of what the proposal actually is.
-This should have enough detail that reviewers can understand exactly what
-you're proposing, but should not include things like API designs or
-implementation. The "Design Details" section below is for the real
-nitty-gritty.
--->
+Tekton is design as an extension of Kubernetes through the [Custom Resource Definition][crd]
+mechanism and, as such, it must abide by the constraints associated with it.
+The most relevant constraint in terms of this TEP is that custom resources (CRDs)
+are defined at the cluster level. As such, CRDs and components that operate on them can only
+be installed once in a Kubernetes cluster, and must be shared across multiple installations
+of Tekton within the same cluster.
+
+In this TEP we propose to separate such components so that it becomes possible for a Tekton
+installer system to install, update and uninstall the parts of Tekton required for multiple
+installation of Tekton to coexist within a single cluster.
+
+The components in questions are:
+- The CRDs themselves. These are defined in dedicated files today, so no action needed for them
```

Comment:
I think the CRDs here refer to `tasks`, `pipelines`, `taskRun`, etc. These CRDs are defined in dedicated files such as https://github.com/tektoncd/pipeline/blob/main/config/300-pipeline.yaml But would these have to be part of a certain namespace and thus change them to include namespace? or am I mixing things up 🙃

Reply:
@pritidesai I "think" you are mixing things up. CRDs are cluster-wide object, they are not attached to a given namespace, they are available on all (the `CR`, may be attached to namespace, depending in the CRD defines the object as namespaced or cluster-wide).